__tests__/Place/place.test.ts

```diff
@@ -0,0 +1,103 @@
+import { Place } from "../../src/db/types";
+import { dbConnection } from "../../src/db/mongoConnection";
+import { getPlaceByID, addPlace } from "../../src/db/Place/place";
+
+jest.setTimeout(15000);
+
+beforeEach(async () => {
+	const { _db, _connection } = await dbConnection();
+	await _db.dropDatabase();
+	await _connection.close();
+
+	try {
+		await addPlace({
+			_id: "faketestid1",
+			avgBraille: null,
+			avgFontReadability: null,
+			avgGuideDogFriendly: null,
+			avgNavigability: null,
+			avgStaffHelpfulness: null,
+		});
+	} catch (e) {
+		console.log(e);
+	}
+
+	try {
+		await addPlace({
+			_id: "faketestid4",
+			avgBraille: null,
+			avgFontReadability: null,
+			avgGuideDogFriendly: null,
+			avgNavigability: null,
+			avgStaffHelpfulness: null,
+		});
+	} catch (e) {
+		console.log(e);
+	}
+});
+
+describe("Place REST endpoints", () => {
+	it("throws error when it tries to get place that doesn't exist in db", async () => {
+		expect.assertions(1);
+		return await getPlaceByID("faketestid2").catch(e =>
+			expect(e).toEqual("Sorry, no place exists with that ID")
+		);
+	});
+	it("gets place", async () => {
+		let place!: Place;
+		try {
+			place = await getPlaceByID("faketestid1");
+		} catch (e) {
+			console.log(e);
+		}
+
+		expect(place).toMatchObject<Place>({
+			_id: "faketestid1",
+			avgBraille: null,
+			avgFontReadability: null,
+			avgGuideDogFriendly: null,
+			avgNavigability: null,
+			avgStaffHelpfulness: null,
+		});
+	});
+	it("adds place to database", async () => {
+		let place!: Place;
+
+		try {
+			place = await addPlace({
+				_id: "faketestid3",
+				avgBraille: null,
+				avgFontReadability: null,
+				avgGuideDogFriendly: null,
+				avgNavigability: null,
+				avgStaffHelpfulness: null,
+			});
+		} catch (e) {
+			console.log(e);
+		}
+
+		expect(place).toMatchObject<Place>({
+			_id: "faketestid3",
+			avgBraille: null,
+			avgFontReadability: null,
+			avgGuideDogFriendly: null,
+			avgNavigability: null,
+			avgStaffHelpfulness: null,
+		});
+	});
+	it("throws error when duplicate place is added", async () => {
+		expect.assertions(1);
+		return await addPlace({
+			_id: "faketestid4",
+			avgBraille: null,
+			avgFontReadability: null,
+			avgGuideDogFriendly: null,
+			avgNavigability: null,
+			avgStaffHelpfulness: null,
+		}).catch(e =>
+			expect(e).toEqual(
+				"That place already exists in the database and cannot be added again."
```

Comment:
remind me, does `toThrow` work with async functions?

Reply:
It did not when I tried, and I tried every way I could find. I should have read the documentation (says every software developer ever) because it said exactly what I needed to do, see here: https://jestjs.io/docs/asynchronous#promises @carpenterd777